### PR TITLE
[chore] make addlicense consistent

### LINKED ISSUE #29863: Difference in addlicense vs checklicense
### Component(s)

_No response_

## Description
In `Makefile.Common`,

For `addlicense`, we provide the following arguments.
```
$(ADDLICENSE) -s=only -y "" -c 'The OpenTelemetry Authors' $(ALL_SRC)
```

`$(ALL_SRC)` enumerates the paths used to add a license, and is defined as

```
ALL_SRC := $(shell find $(ALL_PKG_DIRS) -name '*.go' \
                                -not -path '*/third_party/*' \
                                -not -path '*/local/*' \
                                -type f | sort)
```

However, `checklicense` manually provides it's own list of files as such:

```
.PHONY: checklicense
checklicense: $(ADDLICENSE)
        @licRes=$$(for f in $$(find . -type f \( -iname '*.go' -o -iname '*.sh' \) ! -path '**/third_party/*')
```

Note that `checklicense` will additionally search for shell files, but `addlicense` will not.

## Steps to Reproduce
```
echo '#!/bin/bash' >> receiver/redisreceiver/testdata/integration/will_fail_checklicense.sh
make addlicense
make checklicense
```
## Expected Result
```
❯ rm receiver/redisreceiver/testdata/integration/will_fail_checklicense.sh
❯ make checklicense
    ~/works/otel-g/opentelemetry-collector-contrib    main *9 ?1 
```

## Actual Result
```
❯ echo '#!/bin/bash' >> receiver/redisreceiver/testdata/integration/will_fail_checklicense.sh
make addlicense
make checklicense
Add License finished successfully
license header checking failed:
./receiver/redisreceiver/testdata/integration/will_fail_checklicense.sh
./receiver/redisreceiver/testdata/integration/will_fail_checklicense.sh
make: *** [Makefile.Common:157: checklicense] Error 1
```

### Collector version

v0.91

## Environment
OS:
```
❯ uname -r
5.15.0-89-generic
```

Otel version:
```
❯ git describe --always --match "v[0-9]*" HEAD
v0.91.0-66-g95d75a66ac
```

### OpenTelemetry Collector configuration

_No response_

### Log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by hughesjj**
**Description:** Fixes divergence in add/check license
**Link to tracking Issue:** Resolves #29863

**Testing:**
After this changefix, the reproduction in upstream issue will produce the following output

```
❯ echo '#!/bin/bash' >> receiver/redisreceiver/testdata/integration/will_fail_checklicense.sh
make addlicense
make checklicense
Add License finished successfully
Check License finished successfully
❯ git status
On branch fix_addlicense
Untracked files:
  (use "git add <file>..." to include in what will be committed)
        receiver/redisreceiver/testdata/integration/will_fail_checklicense.sh

nothing added to commit but untracked files present (use "git add" to track)
```

### CODE REVIEW

File: Makefile.Common [hunk 1]
```diff
@@ -81,6 +81,11 @@ ALL_SRC := $(shell find $(ALL_PKG_DIRS) -name '*.go' \
                                 -not -path '*/local/*' \
                                 -type f | sort)
 
+ALL_SRC_AND_SHELL := $(shell find $(ALL_PKG_DIRS) -name '*.go' -o -name "*.sh" \
+                                -not -path '*/third_party/*' \
+                                -not -path '*/local/*' \
```
Comment (crobert-1) [inline Makefile.Common:86]: I noticed `local/` wasn't originally ignored, but found it's in [`.gitignore`,](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/615aad997ab60a8123d5660d61497fce67f16948/.gitignore#L1) so there's no point in including it in makefile logic.

(I agree with your change here, just pointing out why)

File: Makefile.Common [hunk 2]
```diff
@@ -81,6 +81,11 @@ ALL_SRC := $(shell find $(ALL_PKG_DIRS) -name '*.go' \
                                 -not -path '*/local/*' \
                                 -type f | sort)
 
+ALL_SRC_AND_SHELL := $(shell find $(ALL_PKG_DIRS) -name '*.go' -o -name "*.sh" \
+                                -not -path '*/third_party/*' \
```
Comment (crobert-1) [inline Makefile.Common:85]:
```suggestion
                                -not -path '**/third_party/*' \
```
We may want to keep the same regex here for `third_party` directories, it looks like the `**` originally there allows crossing directory boundaries, so any `third_party` sub-directory would be ignored, not just the top level. [Relevant link](https://stackoverflow.com/questions/46547540/meaning-of-a-double-star-in-a-file-path). Let me know if I'm missing something here.

**Review by crobert-1 — CHANGES_REQUESTED** [review]

**Comment by crobert-1** [comment]
For the sake of complete testing, could you compare the list of files returned by the original `checklicense` `find` command, and the new one? You'd probably have to sort the old one (the new one's sorted) to get them to match, but it would be good to ensure at this point there's no change to `checklicense`. You don't have to share the complete lists, but if you could just confirm they match it would be good.

**Comment by crobert-1** [comment]
> Link to tracking Issue: [29863]

One more thing for now, could you change the above line in your description to be:
```
Link to tracking issue:
Resolves #29863
```
This will allow GitHub to link and automatically close the issue when the PR is resolved.

File: Makefile.Common (hunk 2 above)
Comment (crobert-1) [reply Makefile.Common:85]: Fair point, I missed that this was consistent with other references in the makefile. In that case, it's fine with me to leave it for now, but I'll defer to others if there are more concerns here. 👍

File: Makefile.Common [hunk 3]
```diff
@@ -89,6 +91,9 @@ ALL_SRC_AND_DOC := $(shell find $(ALL_PKG_DIRS) -name "*.md" -o -name "*.go" -o
 # ALL_PKGS is used with 'go cover'
 ALL_PKGS := $(shell $(GOCMD) list $(sort $(dir $(ALL_SRC))))
 
+
```
Comment (crobert-1) [inline Makefile.Common:94]:
```suggestion
```

**Review by crobert-1 — APPROVED** [review]

**Review by atoulme — APPROVED** [review]

**Review by pjanotti — APPROVED** [review]

**Pull request merged**